src/highlight/HighlightPromoter.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import classNames from 'classnames';
+import useOutsideEvent from '../common/useOutsideEvent';
+import { getSelectionRange } from './highlightUtil';
+import { MOUSE_PRIMARY } from '../constants';
+import './HighlightPromoter.scss';
+
+type Props = {
+    className?: string;
+    pageEl: HTMLElement;
+};
+
+export default function HighlightPromoter({ className, pageEl }: Props): JSX.Element {
+    const isSelectingRef = React.useRef<boolean>(false);
+    const timerRef = React.useRef<NodeJS.Timeout | null>(null);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const handleRangeChange = (_range: Range | null): void => {
+        // show highlight promotion popover
+    };
+
+    const handleMouseDown = ({ buttons }: MouseEvent): void => {
```

Comment:
Does this all work with touch events? Keyboard shortcuts (Shift+Arrow after selecting text with the mouse, for example)? Should we use the `selectionchange` event API, instead?